### DISCUSSION TITLE
Change line thickness

### DISCUSSION THREAD

**Discussion by FlightBlaze** [DiligentGraphics/DiligentEngine#198] [Q&A] [answered]
I drew a wireframe of line primitives but the lines themselves are too thin. Can I increase the line thickness?

- The first option is to use a 2D graphics library, but we lose the depth.
- The second is to use a geometry shader, but does it work on mobile phones? I will prefer the second option if it is the only option in my case.

Screenshot:

<img width="1440" alt="Screenshot 2022-05-22 at 21 34 18" src="https://user-images.githubusercontent.com/18074076/169710487-aec82cd2-4a2d-43b5-aa89-a4554400b334.png">

**Comment by TheMostDiligent** [comment, marked as answer]
Geometry shaders are part of the OpenGLES3.2 spec, which was released in 2015.
In Vulkan, it is [supported](https://vulkan.gpuinfo.org/listfeaturescore10.php?platform=android) on 63% Android devices.
Wireframe mode itself is not always supported on mobile devices and in many cases it is very slow (as is line rendering), so other methods are preferable.

**Reply by FlightBlaze** [reply]
I made a thick line by extruding a quad along the line.

Screenshot:

<img width="1440" alt="Screenshot 2022-05-23 at 00 48 40" src="https://user-images.githubusercontent.com/18074076/169717374-71e7335c-d9cd-4c2a-9910-c8112a40399d.png">

**Reply by TheMostDiligent** [reply]
Yes, this is certainly one way to do this, and it doesn't require any special capabilities. Most likely this is faster too, or the driver may do the same thing to make a thick line.
